components/resolved_ts/src/observer.rs

```diff
@@ -16,16 +16,24 @@ use crate::endpoint::Task;
 pub struct Observer<E: KvEngine> {
     cmd_batches: RefCell<Vec<CmdBatch>>,
     scheduler: Scheduler<Task<E::Snapshot>>,
+    need_old_value: bool,
 }
 
 impl<E: KvEngine> Observer<E> {
     pub fn new(scheduler: Scheduler<Task<E::Snapshot>>) -> Self {
         Observer {
             cmd_batches: RefCell::default(),
             scheduler,
+            need_old_value: true,
         }
     }
 
+    // Disable old value, currently only use in tests to avoid holding the snapshot
+    // and cause data can not be deleted
+    pub fn disable_old_value(&mut self) {
```

Comment:
If don't delete it, will big impact in the test? I think it is best not to invade the normal code logic.

Reply:
There are few reasons to doing so, firstly we need to start the `resolved-ts` worker thread for our test case, but if we start it by default (for all test), the worker will panic because there are many test writing raw kv data (even directly writing to the transaction cf i.e. `write_cf`, `lock_cf`) and the worker will panic while parsing these raw data. So that we can't start the worker by default.

But even so, the apply worker will still send data to the `resolved-ts` observer, and if old value is enabled there will be a snapshot in the observer task, and since the `resolved-ts` worker is not start, the snapshot will not be dropped and causing data can't delete then some test cases (that expect data to be deleted) will fail.

Reply:
What about using `new_only_for_test` instead of `disable_old_value` ?

Reply:
> What about using new_only_for_test instead of disable_old_value ?

Don't seem necessary to me, `Observer::disable_old_value` and `Observer:: new` are use separately.

Reply:
> the worker will panic while parsing these raw data
you mean we do not support raw kv?

> the snapshot will not be dropped and causing data can't delete then some test cases
why cannot start resolved work deafaultly?

Reply:
> you mean we do not support raw kv?

Currently in the master branch, yes. It is because resolved worker tries to parse raw kv data as txn data,  but it is not by design and will be fixed in the next PR. 

> why cannot start resolved work in default?

"because there are many tests writing raw kv data (even directly writing to the transaction cf i.e. write_cf, lock_cf) and the worker will panic while parsing these raw data."

Some tests writing raw kv data into transaction cf (i.e. `write_cf`, `lock_cf`), it is hard for the resolved worker to distinguish test data and actually invalid data.